src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -348,13 +347,14 @@ public boolean isParent(File folder, File file) {
     }
 
     /**
+     * Returns a <code>File</code> object which is normally constructed with <code>new
+     * File(parent, fileName)</code> except when the parent and child are both
+     * special folders, in which case the <code>File</code> is a wrapper containing
+     * a <code>ShellFolder</code> object.
```

Comment:
Is a "ShellFolder" part of the public API? If not should we mention it in the spec?

Reply:
It is already mentioned in getSystemDisplayName, getSystemTypeDescription, getSystemIcon, isRoot. Anyways, I have removed the linkage for the new addition as it will not be able to resolve.

Reply:
If `ShellFolder` remains mentioned, I'd rather use `<code>ShellFolder</code>` as it was done originally.

I guess Sergey challenged the fact of referencing `ShellFolder`. The fact that it could be `ShellFolder` rather than `File` is an implementation detail. Can we drop that from the spec?

On the other hand, `ShellFolder` class, even though it's not a public API, is mentioned in quite a few methods in `FileSystemView`.

I don't have a strong opinion here. Perhaps, we should scrap all the references to `ShellFolder`.

Reply:
Since it is mentioned in other places, I would rather not scrap it. Also, since in most places it is mentioned as a non-link value, I chose to modify it.

Reply:
The `<code>` element does not create a link, it uses monospace font to render its contents which implies it's part of computer code. Class names usually marked up with `<code>` HTML element or with `{@code }` javadoc element.

These aren't used consistently. Often classes aren't marked up. So you're right: in the majority of cases, `ShellFolder` isn't marked up. The `<code>` markup is used in two methods only: `isRoot` and `getChild`.

Reply:
I was referring to removing all the references to `ShellFolder` from public API, under another bugid, if it's to be done.